x-pack/plugins/observability_solution/inventory/public/hooks/use_detail_view_redirect.ts

```diff
@@ -0,0 +1,117 @@
+import { useCallback } from 'react';
+import { DashboardLocatorParams } from '@kbn/dashboard-plugin/public';
+import { DASHBOARD_APP_LOCATOR } from '@kbn/deeplinks-analytics';
+import { castArray } from 'lodash';
+import { Exception } from 'handlebars';
+import type { Entity } from '../../common/entities';
+import { unflattenEntity } from '../../common/utils/unflatten_entity';
+import { useKibana } from './use_kibana';
+
+const KUBERNETES_DASHBOARDS_IDS: Record<string, string> = {
+  [ENTITY_TYPES.KUBERNETES.CLUSTER.ecs]: 'kubernetes-f4dc26db-1b53-4ea2-a78b-1bfab8ea267c',
+  [ENTITY_TYPES.KUBERNETES.CLUSTER.semconv]: 'kubernetes_otel-cluster-overview',
+  [ENTITY_TYPES.KUBERNETES.CRONJOB.ecs]: 'kubernetes-0a672d50-bcb1-11ec-b64f-7dd6e8e82013',
+  [ENTITY_TYPES.KUBERNETES.DAEMONSET.ecs]: 'kubernetes-85879010-bcb1-11ec-b64f-7dd6e8e82013',
+  [ENTITY_TYPES.KUBERNETES.DEPLOYMENT.ecs]: 'kubernetes-5be46210-bcb1-11ec-b64f-7dd6e8e82013',
+  [ENTITY_TYPES.KUBERNETES.JOB.ecs]: 'kubernetes-9bf990a0-bcb1-11ec-b64f-7dd6e8e82013',
+  [ENTITY_TYPES.KUBERNETES.NODE.ecs]: 'kubernetes-b945b7b0-bcb1-11ec-b64f-7dd6e8e82013',
+  [ENTITY_TYPES.KUBERNETES.POD.ecs]: 'kubernetes-3d4d9290-bcb1-11ec-b64f-7dd6e8e82013',
+  [ENTITY_TYPES.KUBERNETES.STATEFULSET.ecs]: 'kubernetes-21694370-bcb2-11ec-b64f-7dd6e8e82013',
+};
+
+export const useDetailViewRedirect = () => {
+  const {
+    services: { share, entityManager },
+  } = useKibana();
+
+  const locators = share.url.locators;
+  const assetDetailsLocator = locators.get<AssetDetailsLocatorParams>(ASSET_DETAILS_LOCATOR_ID);
+  const dashboardLocator = locators.get<DashboardLocatorParams>(DASHBOARD_APP_LOCATOR);
+  const serviceOverviewLocator = locators.get<ServiceOverviewParams>(SERVICE_OVERVIEW_LOCATOR_ID);
+
+  const getSingleIdentityFieldValue = useCallback(
+    (entity: Entity) => {
+      const identityFields = castArray(entity[ENTITY_IDENTITY_FIELDS]);
+      if (identityFields.length > 1) {
+        throw new Exception(
+          `Multiple identity fields are not supported for ${entity[ENTITY_TYPE]}`
+        );
+      }
+
+      const identityField = identityFields[0];
+      return entityManager.entityClient.getIdentityFieldsValue(unflattenEntity(entity))[
+        identityField
+      ];
+    },
+    [entityManager.entityClient]
+  );
+
+  const getDetailViewRedirectUrl = useCallback(
+    (entity: Entity) => {
+      const type = entity[ENTITY_TYPE];
+      const identityValue = getSingleIdentityFieldValue(entity);
+
+      switch (type) {
+        case ENTITY_TYPES.HOST:
+        case ENTITY_TYPES.CONTAINER:
+          return assetDetailsLocator?.getRedirectUrl({
+            assetId: identityValue,
+            assetType: type,
+          });
+
+        case 'service':
+          return serviceOverviewLocator?.getRedirectUrl({
+            serviceName: identityValue,
+            // service.environemnt is not part of entity.identityFields
+            // we need to manually get its value
```

Comment:
IMO, we don't have to pass the environment if it's not part of the identityFields.

Reply:
The Service Overview expects it though. It's used there to filter the data by environment. If not passed I'm not sure if there will be consequences.

Reply:
`environment` is optional it will probably select "ALL" environments.

Reply:
yes. and if we don't pass it at all, it will always select ALL. Would that be correct?

Reply:
Yes. We do the same for the service inventory. If a service has more than one environments and the top right selector has All we don't pass the environment

Reply:
right. just for context, we were already passing the `service.environment` before this PR. @kpatticha , is the suggestion here to change that and not pass it anymore?

Reply:
Yes. I suspect we were passing the `service.environment` cause initially it was part of the `identityFields`. When we removed it from the entity defintion we didn't update the ui.